navigator_engine/pluggable_logic/conditional_functions.py

```diff
@@ -57,3 +58,21 @@ def check_dataset_valid(engine: DecisionEngine) -> bool:
         if resource.get('validation_status', 'success') != 'success':
             return False
     return True
+
+
+@register_conditional
+def check_spectrum_file(indicators: list[str], engine: DecisionEngine) -> bool:
+    # Loads the CSV as a dataframe as you have stored it in your data loader
+    # Here we have called the data loader with "name" arg = 'spectrum-checker'
+    checklist = engine.data['spectrum-checker']['data']
+
+    if checklist is not None:
+        for indicator in indicators:
+            if indicator not in checklist['Condition checked'].values:
+                raise navigator_engine.common.DecisionError(
+                    f'Indicator "{indicator}" not found in Spectrum checklist'
+                )
+        indicator_checks = checklist[checklist['Condition checked'].isin(indicators)]
+        return indicator_checks['Status'].eq(True).all()
```

Comment:
As I have mentioned in the Jira ticket, this will need to allow various values to evaluate to true.

Reply:
The pandas CSV parser will interpret the value "NA" as pandas nan. I've added a line to fill nan's with True for the dataframe, but it is possible other values will get parsed as nan's as well and therefore treated as True. Do we have a list of values we want to interpret as True and are there some we'd still want to treat as False?